src/utils/TypeManager.h

```diff
@@ -182,7 +184,7 @@ static bool isItselfType(T1* p){
 	if (p==nullptr)
 		return false;
 	const bool convertible = is_convertible<T1*, Object*>::value;
-	typeinfo<convertible>::is_same<T1, T2>(p);
+	typeinfo_same<convertible>::template is_same<T1, T2>(p);
```

Comment:
add template into "::template is_same..." to fix 'expected primary-expression before'.